Sources/Helpers/DirectoryObserver.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  DirectoryObserver.swift
+//  OsmAnd Maps
+//
+//  Created by Skalii on 26.07.2024.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import Foundation
+
+@objc enum DirectoryObserverType: Int {
+    case colorPalette
+
+    var path: String {
+        switch self {
+        case .colorPalette:
+            return OsmAndApp.swiftInstance().colorsPalettePath
+        }
+    }
+
+    var notificationName: NSNotification.Name {
+        switch self {
+        case .colorPalette:
+            return NSNotification.Name("ColorPaletteDicrectoryUpdated")
+        }
+    }
+}
+
+@objcMembers
+final class DirectoryObserver: NSObject {
+
+    static let updatedKey = "updated"
+    static let deletedKey = "deleted"
+    static let createdKey = "created"
+
+    private let type: DirectoryObserverType
+
+    private var directoryFileDescriptor: CInt = -1
+    private var dispatchSource: DispatchSourceFileSystemObject?
+    private let queue = DispatchQueue(label: "DirectoryObserverQueue", attributes: .concurrent)
+
+    init(_ type: DirectoryObserverType) {
+        self.type = type
+    }
+
+    func startObserving() {
```

Comment:
Why you dropped methods that were in the initial test code? `stopObserving`, `deinit` ?

import Foundation

class DirectoryObserver {
    private var directoryFileDescriptor: CInt = -1
    private var dispatchSource: DispatchSourceFileSystemObject?
    private let directoryURL: URL
    
    init(directoryURL: URL) {
        self.directoryURL = directoryURL
    }
    
    func startObserving() {
        // Open the directory
        directoryFileDescriptor = open(directoryURL.path, O_EVTONLY)
        
        guard directoryFileDescriptor != -1 else {
            print("Unable to open directory: \(directoryURL.path)")
            return
        }
        
        // Create the dispatch source
        dispatchSource = DispatchSource.makeFileSystemObjectSource(
            fileDescriptor: directoryFileDescriptor,
            eventMask: .all,
            queue: DispatchQueue.global()
        )
        
        // Set the event handler
        dispatchSource?.setEventHandler { [weak self] in
            guard let self = self else { return }
            print("Directory contents changed")
            self.handleDirectoryChange()
        }
        
        // Set the cancel handler
        dispatchSource?.setCancelHandler { [weak self] in
            guard let self = self else { return }
            close(self.directoryFileDescriptor)
            self.directoryFileDescriptor = -1
        }
        
        // Start monitoring
        dispatchSource?.resume()
    }
    
    func stopObserving() {
        dispatchSource?.cancel()
    }
    
    private func handleDirectoryChange() {
        // Implement your logic to handle directory changes
        print("Handle the changes here")
    }
    
    deinit {
        stopObserving()
    }
}

// Usage
let directoryURL = URL(fileURLWithPath: "/path/to/your/directory")
let observer = DirectoryObserver(directoryURL: directoryURL)
observer.startObserving()